### DISCUSSION TITLE
How exactly do I update the ArchLinux environment?

### DISCUSSION THREAD

**Discussion by wmartinmimi** [TermuxArch/TermuxArch#50] [Q&A] [answered]
hi, im too dumb to understand how to update the archlinux environment.

when TermuxArch updates,
it is clear i need to install the latest version.
so,
to update the environment, do i do :
  1. use setupTermuxArch
  2. use pacman -Syu
  3. both
  4. other ways i don't know?

i hope i can get help,
and the README.md does not says anything clear to this (or i just don't understand)

**Comment by wmartinmimi** [comment]
> hi, im too dumb to understand how to update the archlinux environment.
> 
> when TermuxArch updates,
> it is clear i need to install the latest version.
> so,
> to update the environment, do i do :
>   1. use setupTermuxArch
>   2. use pacman -Syu
>   3. both
>   4. other ways i don't know?
> 
> i hope i can get help,
> and the README.md does not says anything clear to this (or i just don't understand)

also will the update wipe the stuff inside the arch environment?

**Reply by SDRausty** [reply]
>also will the update wipe the stuff inside the arch environment?

Thank you for asking an important question; 
 I have amended information in the help to help answer this question.  Please see `bash setupTermuxArch h` for more information:
>[refresh] [customdir]]  Refresh the Arch Linux in Termux PRoot scripts created by TermuxArch and the installation itself.  Useful for refreshing the installation, the root user's home directory, user home directories and the TermuxArch generated scripts to their newest version;  Directory /var/backups/ backs up the refreshed files.  This refresh mode also runs keys and generates locales.

>[re [customdir]]  Refresh the Arch Linux in Termux PRoot scripts created by TermuxArch.  Useful for refreshing the root user's home directory and user home directories and the TermuxArch generated scripts to their newest version;  Directory /var/backups/ backs up the refreshed files.


>[r [customdir]]  Refresh the Arch Linux in Termux PRoot scripts created by TermuxArch.  Useful for only refreshing the root user's home directory and the TermuxArch generated scripts to their newest version;  Directory /var/backups/ backs up the refreshed files.

**Comment by SDRausty** [comment, marked as answer]
>to update the environment
>use setupTermuxArch

The command `bash setupTermuxArch refresh` can be used ; This command `bash setupTermuxArch h` has more information.

>use `pacman -Syu`

The shortcut command `pci` does this as well.
>both

Yes, you can use both commands.
>other ways i don't know?

You can also build and install from source;  Please ensure that Android, Termux-app and Termux-packages are updated as well.
